src/Doctrine/EntityRegenerator.php

```diff
@@ -66,6 +66,33 @@ public function regenerateEntities(string $classOrNamespace)
                 $classPath = $this->getPathOfClass($classMetadata->name);
             }
 
+            /* @var $classReflection \ReflectionClass */
+            $classReflection = $classMetadata->reflClass;
+
+            $targetFields = array_merge(
+                array_keys($classMetadata->fieldMappings),
+                array_keys($classMetadata->associationMappings)
+            );
+
+            if ($classReflection) {
+                // exclude traits
+                $traitProperties = [];
+
+                foreach ($classReflection->getTraits() as $trait) {
+                    foreach ($trait->getProperties() as $property) {
+                        $traitProperties[] = $property->getName();
+                    }
+                }
+
+                $targetFields = array_diff($targetFields, $traitProperties);
+
+                // exclude inherited properties
+                $targetFields = array_filter($targetFields, function ($field) use ($classReflection) {
+                    return $classReflection->hasProperty($field) &&
+                        $classReflection->getProperty($field)->getDeclaringClass()->getName() == $classReflection->getName();
+                });
+            }
```

Comment:
Let's refactor this into a private method, just for clarity - like `getMappedFieldsInEntity()`.

Reply:
Moved, variable renamed to "$mappedFields"